### PR TITLE
collect form with angular

### PR DESCRIPTION
**Pull request by Yuriy777**

### CODE REVIEW

File: src/app/general-form-bootstrap/form/general-form-bootstrap.component.ts [hunk 1]
```diff
@@ -7,10 +9,207 @@ import { Component, OnInit } from '@angular/core';
...
+    if ( sessionStorage.getItem('form') ) {
+      try {
+        this.form.setValue(JSON.parse(sessionStorage.getItem('form')));
+      } catch (error) {
+
+      }
+    }
+  }
+
+  @HostListener('window:beforeunload', ['$event']) unloadHandler(event: Event) {
```
Comment (Betrozov) [inline src/app/general-form-bootstrap/form/general-form-bootstrap.component.ts]: @Yuriy777 For why do you need `event`?

File: src/app/general-form-bootstrap/form/general-form-bootstrap.component.ts [hunk 2]
```diff
@@ -7,10 +9,207 @@ import { Component, OnInit } from '@angular/core';
...
+    sessionStorage.setItem('form', JSON.stringify(this.form.value));
+  }
+
+  public sendForm(event) {
+    event.preventDefault();
+
+    console.log(this.form.value);
```
Comment (Betrozov) [inline src/app/general-form-bootstrap/form/general-form-bootstrap.component.ts]: @Yuriy777 remove `console`

File: src/app/general-form-bootstrap/form/general-form-bootstrap.component.ts [hunk 3]
```diff
@@ -7,10 +9,207 @@ import { Component, OnInit } from '@angular/core';
...
+    notes5: new FormControl(),
+    notes6: new FormControl(),
+  });
+
   constructor() {
+
   }
 
   ngOnInit() {
+    if ( sessionStorage.getItem('form') ) {
```
Comment (Betrozov) [inline src/app/general-form-bootstrap/form/general-form-bootstrap.component.ts]: @Yuriy777 Why do you use `sessionStorage` but not `localStorage`?

File: src/app/general-form-bootstrap/form/general-form-bootstrap.component.ts [hunk 4]
```diff
@@ -7,10 +9,207 @@ import { Component, OnInit } from '@angular/core';
...
+      try {
+        this.form.setValue(JSON.parse(sessionStorage.getItem('form')));
+      } catch (error) {
```
Comment (Betrozov) [inline src/app/general-form-bootstrap/form/general-form-bootstrap.component.ts]: @Yuriy777 please add error handler

File: src/app/general-form-bootstrap/form/general-form-bootstrap.component.ts [hunk 5]
```diff
@@ -7,10 +9,207 @@ import { Component, OnInit } from '@angular/core';
...
+  });
+
   constructor() {
+
   }
 
   ngOnInit() {
+    if ( sessionStorage.getItem('form') ) {
+      try {
+        this.form.setValue(JSON.parse(sessionStorage.getItem('form')));
```
Comment (Betrozov) [inline src/app/general-form-bootstrap/form/general-form-bootstrap.component.ts]: @Yuriy777 only `JSON.parse` has to be wrapped in `try ... catch`

File: src/app/general-form-bootstrap/form/general-form-bootstrap.component.ts [hunk 6]
```diff
@@ -7,10 +9,207 @@ import { Component, OnInit } from '@angular/core';
...
+
+    // Notes, Recon Cost and Special Equipment Comments
+    notes2: new FormControl(),
+    notes3: new FormControl(),
+    notes4: new FormControl(),
+    notes5: new FormControl(),
+    notes6: new FormControl(),
+  });
+
   constructor() {
```
Comment (Betrozov) [inline src/app/general-form-bootstrap/form/general-form-bootstrap.component.ts]: @Yuriy777 remove empty `constructor`

File: src/app/general-form-material/form/general-form-material.component.ts [hunk 7]
```diff
@@ -1,4 +1,5 @@
 import { Component, OnInit } from '@angular/core';
```
Comment (Betrozov) [inline src/app/general-form-material/form/general-form-material.component.ts]: @Yuriy777 please remove everything that related to `material`

File: package-lock.json [hunk 8]
```diff
@@ -6530,6 +6530,11 @@
       "integrity": "sha1-yobR/ogoFpsBICCOPchCS524NCw=",
       "dev": true
     },
+    "ngx-bootstrap": {
```
Comment (Betrozov) [inline package-lock.json]: @Yuriy777 Do you need it?

File: src/app/general-form-bootstrap/form/general-form-bootstrap.component.ts (hunk 2 above)
Comment (Yuriy777) [reply src/app/general-form-bootstrap/form/general-form-bootstrap.component.ts]: I want to show on demo that form working

File: src/app/general-form-bootstrap/form/general-form-bootstrap.component.ts (hunk 3 above)
Comment (Yuriy777) [reply src/app/general-form-bootstrap/form/general-form-bootstrap.component.ts]: It is wrote in the mail which  Marina sent to Randall after first call

File: src/app/general-form-bootstrap/form/general-form-bootstrap.component.ts (hunk 4 above)
Comment (Yuriy777) [reply src/app/general-form-bootstrap/form/general-form-bootstrap.component.ts]: When I will have error, I upload clean form without any data

File: src/app/general-form-bootstrap/form/general-form-bootstrap.component.ts (hunk 5 above)
Comment (Yuriy777) [reply src/app/general-form-bootstrap/form/general-form-bootstrap.component.ts]: When I will have error, I upload clean form without any data. If I wrap only JSON.parse in try ... catch I will have error in setValue method

**Pull request merged**